src/resolve/resolved-dns-cache.c

```diff
@@ -178,18 +174,21 @@ static void dns_cache_make_space(DnsCache *c, unsigned add) {
                 return;
 
         /* Makes space for n new entries. Note that we actually allow
-         * the cache to grow beyond CACHE_MAX, but only when we shall
-         * add more RRs to the cache than CACHE_MAX at once. In that
+         * the cache to grow beyond max_size, but only when we shall
+         * add more RRs to the cache than max_size at once. In that
          * case the cache will be emptied completely otherwise. */
 
         for (;;) {
                 _cleanup_(dns_resource_key_unrefp) DnsResourceKey *key = NULL;
                 DnsCacheItem *i;
+                unsigned size;
 
                 if (prioq_isempty(c->by_expiry))
                         break;
 
-                if (prioq_size(c->by_expiry) + add < CACHE_MAX)
+                /* Check for overflow in case size is close to UINT_MAX. */
+                size = prioq_size(c->by_expiry);
```

Comment:
Please use size_t, even prioq_size() returns unsigned. Also, not a matter, but nowadays we allow to declare variables when necessary. Hence,
```c
size_t size = prioq_size(c->by_expiry);
```